codebuild/codebuild_apply_terraform/code_build_terraform.yml

```
@@ -13,49 +13,49 @@ phases:
       - copy_artifacts.py -j $COPY_ARTIFACTS
       - REPO=$(pwd | awk -F '/' '{print $NF}') # For the artefact name
       - ACTIONS_REQUIRED="${CODEBUILD_SRC_DIR_actions_required}/actions_required.json"
-      - if [ -n "$CHECK_TRIGGER" ]; then
+      - if [ -z  "$CHECK_TRIGGER" ] || [ "$CHECK_TRIGGER" -eq 0 ]; then
+         echo "CHECK_TRIGGER not supplied..";
+         TRIGGER_BUILD=0;
+        elif [ ! -z "$CHECK_TRIGGER" ] || [ "$CHECK_TRIGGER" -eq 1 ]; then
```

Since `||` is used outside of the `[ ]` expression, the return value of the expression is [going to be that of the last command in the list](https://www.gnu.org/software/bash/manual/html_node/Lists.html#Lists), i.e. `[ "$CHECK_TRIGGER" -eq 0 ]` or `[ "$CHECK_TRIGGER" -eq 1 ]`. This could cause a false negative if `CHECK_TRIGGER` is not 0 or 1 but it is a non-zero length string.

```suggestion
      - if [ -z  "$CHECK_TRIGGER"  ||  "$CHECK_TRIGGER" -eq 0 ]; then
         echo "CHECK_TRIGGER not supplied..";
         TRIGGER_BUILD=0;
        elif [ ! -z "$CHECK_TRIGGER"  ||  "$CHECK_TRIGGER" -eq 1 ]; then
```

---

Latest commit hopefully fixes the broken logic.

Tested it in a basic shell script -:

```
if [[ -z "$CHECK_TRIGGER" || "$CHECK_TRIGGER" -eq 0 ]]; then
    echo "CHECK_TRIGGER is either not set or set to 0"
    echo "No TF to validate or deploy"
    NO_TF_VALIDATION=1
elif [[ ! -z "$CHECK_TRIGGER" && "$CHECK_TRIGGER" -eq 1 ]]; then
    echo "CHECK_TRIGGER is set and is 1"
    TRIGGER_BUILD=1
fi

if [ ! -z "$NO_TF_VALIDATION" ]; then
    echo "Nothing to validate"
elif [ ! -z "$TRIGGER_BUILD" ]; then
    echo "Trigger build, run TF validation etc"
elif [ -z "$TRIGGER_BUILD" ]; then
    echo "do nothing"
fi
```

```
bash-3.2$ env | grep CHECK_TRIGGER
bash-3.2$ ./aaron.sh 
CHECK_TRIGGER is either not set or set to 0
No TF to validate or deploy
Nothing to validate
bash-3.2$ export CHECK_TRIGGER=1
bash-3.2$ ./aaron.sh 
CHECK_TRIGGER is set and is 1
Trigger build, run TF validation etc
bash-3.2$ export CHECK_TRIGGER=0
bash-3.2$ ./aaron.sh 
CHECK_TRIGGER is either not set or set to 0
No TF to validate or deploy
Nothing to validate
```